modules/analysis/src/service/load.rs

```diff
@@ -237,9 +248,59 @@ impl AnalysisService {
                 .distinct()
                 .into_query(),
             GraphQuery::Query(query) => sbom_node::Entity::find()
-                .filtering(query.clone())?
+                .join(JoinType::Join, sbom_node::Relation::Package.def())
```

Comment:
I can see why we want to narrow down the set of sboms first ... though concerned about performance ... does this set of joins perform well ?